src/core/reactor_backend.cc

```diff
@@ -1475,46 +1511,39 @@ class reactor_backend_uring final : public reactor_backend {
     }
     virtual future<size_t> read_some(pollable_fd_state& fd, void* buffer, size_t len) override {
         return readable(fd).then([this, &fd, buffer, len] () mutable {
```

Comment:
I think this pessimizes the single-connection case. readable() will return true (speculated), then we queue a request into the ring, but we can have latency of 0.5ms even though data is ready because the reactor will prefer to run tasks and gather more I/O. It's also suboptimal for many connections, since if we speculate that fd is not readable, we'll go through two trips into uring, one for readable() and one for queue_request().

I think we need this logic:

1. check speculation on fd
2. if we speculated that fd is readable, call fd.fd.read() to copy data immediately and return it
3. if we speculated that fd is not readable, call queue_request() immediately

For this to work well, we have to disable SOCK_NONBLOCK when using io_uring, and enable MSG_NOWAIT in step 2. With these changes, we'll have the following:

1. request/reply sockets (data usually not available) will usually wait in IORING_OP_RECV
2. streaming sockets will never wait and use recvmsg(..., MSG_NOWAIT) to copy data from the kernel to the buffer